### PR TITLE
style: add warning to eslint for importing moment

### PR DESCRIPTION
**Pull request by borisonr**
## Overview
Warn people from adding more moment to the codebase and try to create more transparency for our migration to date-fns.
(not sure if `style` is appropriate for this PR name, so let me know if it should be something else)

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [GROW-1929](https://codecademy.atlassian.net/browse/GROW-1929)
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/eslint-config/index.js [hunk 1]
```diff
@@ -116,9 +116,19 @@ module.exports = {
     '@typescript-eslint/quotes': 'off',
 
     'no-restricted-imports': [
-      'error',
+      'warn',
```
Comment (borisonr) [inline packages/eslint-config/index.js:119]: got permission form @codecaaron to change this to warn and will change it back to error after we finish the migration

Comment (JoshuaKGoldberg) [reply packages/eslint-config/index.js:119]: meh, I'd rather just `'error'` and get up in people's faces 😛  either works for me

**Review by JoshuaKGoldberg — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.0.3-alpha.74bf31.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f8087efb126c320e3167541--gamut-preview.netlify.app>https://5f8087efb126c320e3167541--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f8087efb126c320e3167541>Deploy Logs</a></p>

File: packages/eslint-config/index.js (hunk 1 above)
Comment (borisonr) [reply packages/eslint-config/index.js:119]: lol but then the `verify_linting_script` job will fail for everyone...

Comment (borisonr) [reply packages/eslint-config/index.js:119]: or you want me to add the eslint-disable to every reference??

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.0.3-alpha.9ab402.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f80899fa43ced209b60aa15--gamut-preview.netlify.app>https://5f80899fa43ced209b60aa15--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f80899fa43ced209b60aa15>Deploy Logs</a></p>

File: packages/eslint-config/index.js (hunk 1 above)
Comment (codecaaron) [reply packages/eslint-config/index.js:119]: This is an i[nteresting possible solution](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-restricted-paths.md) to a tiered approach and cover our environment use cases here: 
https://github.com/codecademy-engineering/Codecademy/blob/master/webpack/assets/state/.eslintrc.json

Comment (JoshuaKGoldberg) [reply packages/eslint-config/index.js:119]: > add the eslint-disable to every reference

Actually yes - that way people don't look at the imports and assume it's something they can copy & paste

Comment (borisonr) [reply packages/eslint-config/index.js:119]: > > add the eslint-disable to every reference
> 
> Actually yes - that way people don't look at the imports and assume it's something they can copy & paste

but there will be a yellow squiggly line when they look at it!

Comment (JoshuaKGoldberg) [reply packages/eslint-config/index.js:119]: > be a yellow squiggly line

If they're using the ESLint extension. Also, IME even when people do install it, a lot will ignore it.

I don't think this is a blocker at all given how short lived the migration period will be, by the way. This is just an old pain point / hill to die on for me 😂 .

**Pull request merged**